torch/nn/parameter.py

```diff
@@ -44,3 +44,93 @@ def __reduce_ex__(self, proto):
         )
 
     __torch_function__ = _disabled_torch_function_impl
+
+
+class UninitializedParameter(Parameter):
+    r"""A parameter that can be lazily initialized.
```

Comment:
It's not that the parameter can be lazy initialized so much as the uninitialized parameter represents a parameter that is not yet initialized.